### ISSUE TITLE
Is there any way to ignore fields?

### ISSUE THREAD

**Issue by lucasrpb** [scala/pickling#141]
I would like to know if there's a way to ignore class fields. Something like this:

case class test(@transient var x: Int, var y: String)

I would want ignore x var on serialisation. Is it possible?

**Comment by btbvoy** [comment]
Here is a failing case which isn't working.
**Given**:

``` scala
case class Person(name: String, @transient age: Int)
val p = Person("Simon", 20).pickle
println(p.value)
```

**Actual**:

``` json
{
  "$type": "this.Person",
  "name": "Simon",
  "age": 20
}
```

**Expected**:

``` json
{
  "$type": "this.Person",
  "name": "Simon"
}
```

Could someone take a look at this case?

Thanks in advance,
Pavel

**Comment by jsuereth** [comment]
FYI -

What you're asking is something I actaully consider somewhat bad practice for case classes:

``` scala
case class Person(name: String, @transient age: Int)
```

Here, you're really violating the spirit of the case class, that is ALL the data placed into the constructor can be reconsituted at runtime via the deconstructor (pattern matching).  If you fail to pickle any field, you actually loose out on the ability to pickle the thing statically and have to resort to runtime reflection.

That said, I have fixed this on the 0.11.x branch.  We can now, appropriately, pay attention to transient fields.   You'll get the expected behavior you want.  However, this will NOT work if you `import static._` because we have no public constructor for a case class with transient argument.

Here is a secnario we CAN handle statically though:

``` scala
case class Foo(x: Int) {
   @transient var y = x + 1
}
```

SO, TL;DR; this is fixed in 0.11.x, but be warned of using `@transient` on `case class`

**Comment by btbvoy** [comment]
awesome news :+1: Thanks for update :smile:
